serial_interface:
  ros__parameters:
    serial_params:
      use_port: True        # if use_port is False, device_hwid will be used to scan all ports
      port: "/dev/ttyACM0"  # has priority over device_hwid
      baud: 115200
      device_hwid: "75735353238351A0A2E1"
    incoming_message:
      header: "SYNCSYNC"
      footer: "/n"
      packet_size: 50
      rate_hz: 10
    publish_topic:
      name: "serial_data"
      frame_id: ""


      # payload:
      #   content: ["frequency", "zero_direction", "amplitude", "phase", "frequency", "zero_direction", "amplitude", "phase", "frequency", "zero_direction", "amplitude", "phase", "frequency", "zero_direction", "amplitude", "phase"]
      #   var_lens: [4, 2, 2, 2, 4, 2, 2, 2, 4, 2, 2, 2, 4, 2, 2, 2]
      # packet: "SYNCSYNC" + "DATA" + "/n": 8 + 40 + 2 = 50 bytes
      # data: frequency, zero_direction, amplitude, phase
      # 4 x {float (4), int (2), int (2), float (2)}: 40 bytes 